### PR TITLE
allow to set channels from the yml file

### PR DESCRIPTION
**Pull request by dpshelio**
I've just needed such option (as used in travis) for sunpy/sunpy#1629.

I'm not sure whether that's they way and I don't know how I could test this locally...

### CODE REVIEW

File: appveyor/install-miniconda.ps1 [hunk 1]
```diff
@@ -76,8 +76,12 @@ $env:PATH = "${env:PYTHON};${env:PYTHON}\Scripts;" + $env:PATH
 
 # Conda config
 conda config --set always_yes true
-conda config --add channels astropy-ci-extras
 
+if ($conda_channels -ne $null) {
```
Comment (bsipocz) [inline appveyor/install-miniconda.ps1]: please use upper case global variable names. Also ``-ne $null`` is not needed I think, if a variable is set it should be ``True`` in the conditional (see examples below for ``ASTROPY_VERSION``).

**Comment by bsipocz** [comment]
Thanks @dpshelio. I'll leave a few inclines.

It will need testing, you could add the CONDA_CHANNELS to ``appveyor.yml`` for one of the builds. 
Also please mention it in README (you can just copy over from the travis part).

File: appveyor/install-miniconda.ps1 (hunk 1 above)
Comment (bsipocz) [inline appveyor/install-miniconda.ps1]: Also it seems to me that this conditional should be the other way around, as it goes into the ``else`` part if no ``conda_channels`` was provided

Comment (bsipocz) [inline appveyor/install-miniconda.ps1]: so what about copying over the logic from ``travis/setup_dependencies_common.sh``. If it's not given set it to be equal of ``astropy-ci-extras astropy``, and then loop over and add all the channels.

**Comment by dpshelio** [comment]
Thanks @bsipocz! I did indeed have the `if` logic the other way around. I've updated the PR as you suggested.

**Comment by Cadair** [comment]
@dpshelio it broke. :(

File: appveyor/install-miniconda.ps1 [hunk 2]
```diff
@@ -76,8 +76,12 @@ $env:PATH = "${env:PYTHON};${env:PYTHON}\Scripts;" + $env:PATH
 
 # Conda config
 conda config --set always_yes true
-conda config --add channels astropy-ci-extras
 
+if (! $env:CONDA_CHANNELS) {
+   conda config --add channels astropy astropy-ci-extras
```
Comment (dpshelio) [inline appveyor/install-miniconda.ps1]: @bsipocz - Ive also added `astropy` as a default channel as done for travis.

Comment (bsipocz) [inline appveyor/install-miniconda.ps1]: Thanks, looks good. Can you actually add multiple channels in the same command?

Comment (Cadair) [inline appveyor/install-miniconda.ps1]: I didn't think you could.

Comment (dpshelio) [inline appveyor/install-miniconda.ps1]: Ahhh! that must be the error....

Comment (Cadair) [inline appveyor/install-miniconda.ps1]: you need

```
conda config --add channels astropy --add channels astropy-ci-extras
```

Comment (bsipocz) [inline appveyor/install-miniconda.ps1]: Me neither, but I never tried it either (Tom wrote this part for travis, so I guess he had reasons to loop over the channels). We'll see in 30 mins.

Comment (bsipocz) [inline appveyor/install-miniconda.ps1]: @Cadair - But that logic wont work for the CONDA_CHANNELS variable, as we don't know how many elements are in it.

File: appveyor/install-miniconda.ps1 [hunk 3]
```diff
@@ -76,7 +76,15 @@ $env:PATH = "${env:PYTHON};${env:PYTHON}\Scripts;" + $env:PATH
...
-conda config --add channels astropy-ci-extras
+
+if (! $env:CONDA_CHANNELS) {
+   $CONDA_CHANNELS=@("astropy", "astropy-ci-extras")
+} else {
+   $CONDA_CHANNELS=$env:CONDA_CHANNELS.split(" ")
+}
+foreach ($CONDA_CHANNEL in $CONDA_CHANNELS) {
+   conda config --add $CONDA_CHANNEL
+}
```
Comment (dpshelio) [inline appveyor/install-miniconda.ps1]: all right... this should work... (I hope)

**Comment by Cadair** [comment]
MOAR BUILDS! D:

**Comment by bsipocz** [comment]
@Cadair - No more builds, I've cancelled the middle ones.

**Comment by Cadair** [comment]
@bsipocz Winning!

**Comment by dpshelio** [comment]
Why????

**Comment by dpshelio** [comment]
Ok, hopefully now it runs well... (I forgot to add `channels` to the `conda config --add`)

**Comment by dpshelio** [comment]
It worked!!!! Yayyy!!!

**Comment by Cadair** [comment]
REJOICE! Our CI can work again!

**Comment by bsipocz** [comment]
Hurray! Thanks @dpshelio!

**Comment by dpshelio** [comment]
Thanks @bsipocz!!

**Pull request merged**